tokens/src/commands.rs

```diff
@@ -54,9 +54,9 @@ use {
 
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
 pub struct Allocation {
```

Comment:
Since this is technically part of the public interface, let's add a new typed Allocation struct instead of changing this one. Also, the original `Allocation` may still be useful for csv serialization/deserialization... (in tests, at least)

Reply:
Hmm, didn't think of it as "public API" ... we also want to preserve `pub fn` signatures then (eg. `pub fn build_spl_token_instructions(`), 

switching everyithing else to `AllocationV2`: https://github.com/solana-labs/solana/pull/33926/commits/b9c5a07dde0f9246197697ad1785fb1494197a65

Reply:
Actually, I am going to sneakily unpub the token fn signatures that take `Allocation` (https://github.com/solana-labs/solana/pull/33937); then you can update them to take the new Allocation type. They are only `pub` by mistake. crates.io reports that no one is using solana-tokens as a dependency, so it shouldn't break anyone. By that argument, we could just change the type signature of Allocation as well, but as I said I think the original struct could actually be helpful for serde.

Reply:
Should be done now: https://github.com/solana-labs/solana/pull/33926/commits/a77649e051b57d8beabccebd1c553f6c5ab3686f